doc/source/cluster/kubernetes/references.md

```diff
@@ -10,8 +10,12 @@ refer to the [API reference][APIReference].
 ## KubeRay API compatibility and guarantees
 
 v1 APIs in the KubeRay project are stable and suitable for production environments.
-While KubeRay maintainers strive to maintain backward compatibility, they reserve the right to deprecate API fields.
-They will clearly communicate the deprecation status, and may remove the APIs after a minimum of two minor releases.
-This policy allows for the necessary evolution of the API while providing a reasonable transition period for users.
+Fields in the v1 APIs will never be removed to preserve compatibility.
```

Comment:
We should be able to remove fields during major version bump? Is `never` a too strong guarantee?

Reply:
Good point, we can break the API however we want if we introduce a v2 API (Kai-Hsun and I actually discussed this earlier today). I think this is implied when saying "Fields in the v1 APIs". Let me know if you think I shoud make it clearer

Reply:
Yea, let's make it a bit clearer that as long as we stay with V1 APIs, fields won't be removed but they can be removed if we decide to do V2 APIs.

Reply:
Updated with:

```
Future major versions of the API (i.e. v2) may have breaking changes and fields removed from v1.
```